docs/book/src/reference/raising-events.md

```diff
@@ -0,0 +1,105 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function as they allow users or any automated processes to see what is going on with a particular object and respond to them.
+
+ Recent Events for an object can be viewed by running `$ kubectl describe <resource kind> <resource name>`. Also, they can be checked by running `$ kubectl get events`. 
+
+<aside class="warning">
+<h1>Events should be raised in certain circumstances only</h1>
+
+Be aware that it is **not** recommended to emit Events for all operations. If authors raise too many events, it brings bad UX experiences for those consuming the solutions on the cluster, and they may find it difficult to filter an actionable event from the clutter. For more information, please take a look at the [Kubernetes APIs convention][Events]. 
+
+</aside>
+
+## Writing Events
+
+Anatomy of an Event:
+
+```go
+Event(object runtime.Object, eventtype, reason, message string)
+```
+
+- `object` is the object this event is about.
+- `eventtype` is this event type, and is either *Normal* or *Warning*. ([More info][Event-Example])
+- `reason` is the reason this event is generated. It should be short and unique with `UpperCamelCase` format. The value could appear in *switch* statements by automation. ([More info][Reason-Example])
+- `message` is intended to be consumed by humans. ([More info][Message-Example])
+
+
+
+<aside class="note">
+<h1>Example Usage</h1>
+
+Following is an example of a code implementation that raises an Event.
+
+```go
+	// The following implementation will raise an event
+	r.Recorder.Event(cr, "Warning", "Deleting",
+		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
+			cr.Name,
+			cr.Namespace))
+```
+
+</aside>
+
+## How to be able to raise Events?
+
+Following are the steps with examples to help you raise events in your controller's reconciliations. 
+Events are published from a Controller using an [EventRecorder]`type CorrelatorOptions struct`,
```

Comment:
It is missing the link:

<img width="545" alt="Screenshot 2023-06-13 at 06 34 19" src="https://github.com/kubernetes-sigs/kubebuilder/assets/7708031/4181d538-056e-4171-9fc1-169ef05f6e2f">

Reply:
Should I add [this](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events) or any other link